docs/optimizations/FileLocalization.md

```diff
@@ -0,0 +1,52 @@
+# The 'localization_optional' Optimization
+
+## Scope
+
+The 'localization_optional' optimization applies to `File` and `File?` inputs on `task`s. It allows
+you to save time and money by perhaps not localizing some `File`s to the execution environment.
+
+## Effect
+
+The optimization signals to Cromwell that the task has been written in such a way that:
+ * The task will still work if the file were to be localized - say from a cloud bucket to the task's execution environment; but...
+ * The task will *also* work if the file is *not* localized prior to execution - and remains a path to a value in an object store, for example.  
+
+Therefore, Cromwell will (usually) choose not to localize this file, in order to save time and money on disk size and `File` localization.
+
+## Language Support
+
+### WDL 1.0 (or later)
+
+In a WDL 1.0 `task`, this optimization is specified by adding a `localization_optional` field to 
+an input's entry in the task's `parameter_meta` section. Here's an example:
+```wdl
+task nio_task {
+  input {
+    File foo_file
+  }
+  
+  parameter_meta {
+    foo_file: {
+      localization_optional: true
+    }
+  }
+  
+  command <<<
+    # This tool must work for **BOTH** local file paths **AND** object store URL values:
+    java -jar my_tool.jar ~{foo_file}
+  >>>
+}
+```
+
+## Backend Support
+
+This optimization is currently only applied to localization in the Pipelines API (GCE) backends.
+
+## Warning
+
+Be very careful not to write unportable WDL files!
+
+* Your task should always be written bearing in mind that a `File` tagged for this optimization
+ *might* still be localized at the whim of the execution engine. 
+* Furthermore, if your WDL is intended to be shared with some other group, that group might 
```

Comment:
Suggested rephrase:
> * If you share your WDL it may be run on engines other than Cromwell, and that engine may not recognize this hint.

What is a hint? I think it is a new concept in Cromwell-WDL-land, I think it warrants a sentence or two explanation.